contracts/staking/compound/GoodCompoundStaking.sol

```diff
@@ -81,34 +81,35 @@ contract GoodCompoundStaking is SimpleStaking {
 	/**
 	 * @dev Function to redeem cToken for DAI, so reserve knows how to handle it. (reserve can handle dai or cdai)
 	 * @dev _amount of token in iToken
+	 * @dev _recipient recipient of the DAI
 	 * @return return address of the DAI and amount of the DAI
 	 */
-	function redeemUnderlyingToDAI(uint256 _amount)
+	function redeemUnderlyingToDAI(uint256 _amount, address _recipient)
 		internal
 		override
 		returns (address, uint256)
 	{
 		ERC20 comp = ERC20(nameService.getAddress("COMP"));
 		uint256 compBalance = comp.balanceOf(address(this));
 		address daiAddress = nameService.getAddress("DAI");
-		Uniswap uniswapContract = Uniswap(
-			nameService.getAddress("UNISWAP_ROUTER")
-		);
+		Uniswap uniswapContract =
+			Uniswap(nameService.getAddress("UNISWAP_ROUTER"));
 		uint256 daiFromComp;
 		cERC20 cToken = cERC20(address(iToken));
 		address[] memory path = new address[](2);
-		address reserveAddress = nameService.getAddress("RESERVE");
 		if (compBalance > 0) {
 			path[0] = address(comp);
 			path[1] = daiAddress;
-			uint256[] memory compSwap = uniswapContract
-			.swapExactTokensForTokens(
-				compBalance,
-				0,
-				path,
-				address(iToken) == nameService.getAddress("CDAI") ? address(this) : reserveAddress,
-				block.timestamp
-			);
+			uint256[] memory compSwap =
+				uniswapContract.swapExactTokensForTokens(
+					compBalance,
+					0,
+					path,
+					address(iToken) == nameService.getAddress("CDAI")
+						? address(this)
+						: _recipient,
+					block.timestamp
```

Comment:
i think we can transfer always to reserve, because in fundmanager we record the starting state of dai+cdai